aip/auth/4115.md

```diff
@@ -5,124 +5,110 @@ state: approved
 ```
 
-In response, the auth library will usually receive a token in the following JSON
-format:
+The request to the MDS endpoint **should** be a HTTP GET request without any parameter or payload.
 
-```json
-{
-      "access_token":"YOUR_ACCESS_TOKEN",
-      "expires_in":3599,
-      "token_type":"Bearer"
- }
-```
+The response from the MDS endpoint **should** contain the following information:
 
-#### Scopes
-Access tokens obtained from the metadata server contain the scopes specified at instance creation.
-For example, if an instance is granted only the
-https://www.googleapis.com/auth/storage-full scope for Cloud Storage, then the tokens obtained 
-from the instance's metadata server have only the `storage-full` scope.
+* The **Secure Session Agent** address: the client doesn’t have direct access to mTLS credentials. 
+  The Secure Session Agent manages default mTLS credentials. The client can only use mTLS 
+  credentials through Secure Session Agent. The address can be an IP:port address or a Unix Domain 
+  Socket (UDS).
 
-When running on GCE, the metadata server ignores requests for custom scopes.
-On GCE equivalent runtimes, clients can request a different set of scopes to the metadata server
-using the `scopes` url parameter, e.g.:
+The client **must** follow the steps below to access Google’s default mTLS credentials.
 
-```
-http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token?scopes=comma-separated-list-of-scopes
-```
-
-The auth library **should** allow the caller to optionally specify a list of custom scopes,
-and add the `scopes` parameter to the request when needed.
-Depending on the runtime environment, the request for custom scopes will be transparently
-ignored by the server (GCE) or fulfilled (GCE equivalent).
+1. Check if the remote endpoint supports mTLS. 
+   * If yes, go to step (2).
+   * If not, go to step (3). 
+2. Send a request to the MDS mTLS configuration endpoint. If the request is successful and the 
+   response contains a Secure Session Agent address, use the address to access Google default mTLS
+   credentials, and go to step (4). If the request fails or the response contains an empty address,
+   go to step (3).
+3. Fall back to TLS [END].
+4. Configure the TLS library to use the Secure Session Agent ([example][3]) for client authentication
+   during the mTLS handshake.
 
-### Google App Engine Standard 1.0 Runtime
+### Request Bound Tokens
 
-If the application runs on a App Engine Standard 1st generation instance, the
-auth library **should** rely on the [Google App Engine SDK][3] to retrieve the
-access token. Here is [one example][5] in Go. Essentially the SDK relies on the
-underlying App Engine Identity service to generate the token.
+To access Google APIs with bound tokens, the client **should** request tokens from MDS. All virtual 
+environments MDS expose a HTTP endpoint (access token endpoint) that serves access tokens.
```

Comment:
micro-nit: add ' at the end of "environments"

Reply:
Done.